modules/social_features/social_user/src/EdaHandler.php

```diff
@@ -0,0 +1,209 @@
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\social_eda\Types\Address;
+use Drupal\social_eda\Types\Application;
+use Drupal\social_eda\Types\DateTime;
+use Drupal\social_eda\Types\Href;
+use Drupal\social_eda\Types\User;
+use Drupal\social_eda_dispatcher\Dispatcher as SocialEdaDispatcher;
+use Drupal\social_user\Event\UserEventData;
+use Drupal\user\UserInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Handles hook invocations for EDA related operations of the user entity.
+ */
+final class EdaHandler {
+
+  /**
+   * The current logged-in user.
+   *
+   * @var \Drupal\user\UserInterface|null
+   */
+  protected ?UserInterface $currentUser = NULL;
+
+  /**
+   * The source.
+   *
+   * @var string
+   */
+  protected string $source;
+
+  /**
+   * The current route name.
+   *
+   * @var string
+   */
+  protected string $routeName;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(
+    private readonly ?SocialEdaDispatcher $dispatcher,
+    private readonly UuidInterface $uuid,
+    private readonly RequestStack $requestStack,
+    private readonly ModuleHandlerInterface $moduleHandler,
+    private readonly EntityTypeManagerInterface $entityTypeManager,
+    private readonly AccountProxyInterface $account,
+    private readonly RouteMatchInterface $routeMatch,
+  ) {
+    // Load the full user entity if the account is authenticated.
+    $account_id = $this->account->id();
+    if ($account_id && $account_id !== 0) {
+      $user = $this->entityTypeManager->getStorage('user')->load($account_id);
+      if ($user instanceof UserInterface) {
+        $this->currentUser = $user;
+      }
+    }
+
+    // Set source.
+    $request = $this->requestStack->getCurrentRequest();
+    $this->source = $request ? $request->getPathInfo() : '';
+
+    // Set route name.
+    $this->routeName = $this->routeMatch->getRouteName() ?: '';
+  }
+
+  /**
+   * Create event handler.
+   */
+  public function userCreate(UserInterface $user): void {
+    $event_type = 'com.getopensocial.cms.user.create';
+    $topic_name = 'com.getopensocial.cms.user.create';
+    $this->dispatch($topic_name, $event_type, $user);
+  }
+
+  /**
+   * Transforms a NodeInterface into a CloudEvent.
+   *
+   * @throws \Drupal\Core\Entity\EntityMalformedException
+   * @throws \Drupal\Core\TypedData\Exception\MissingDataException
+   */
+  public function fromEntity(UserInterface $user, string $event_type): CloudEvent {
+    // Determine actors.
+    [$actor_application, $actor_user] = $this->determineActors();
+
+    // Query for the user's profile based on the user ID and profile type.
+    $profileStorage = $this->entityTypeManager->getStorage('profile');
+
+    /** @var \Drupal\profile\Entity\ProfileInterface[] $profiles */
+    $profiles = $profileStorage->loadByProperties([
```

Comment:
You are using `social_user_profile_insert()` to trigger the event, and I think the profile is not saved to storage at the moment, so `$profiles` will be empty...

Reply:
I checked and it works fine - `social_user_profile_insert()` triggers after the profile is saved.